easybuild/framework/easyblock.py

```diff
@@ -1716,7 +1716,11 @@ def inject_module_extra_paths(self):
             }
 
             try:
-                env_var_opts.update(extra_opts)
+                # make sure paths is added for empty string
+                if extra_opts == '':
+                    env_var_opts['paths'] = extra_opts
+                else:
+                    env_var_opts.update(extra_opts)
```

Comment:
This works, but this `try..else` is really not appropriate given the that `update` does not raise a `ValueError` on empty strings.

Can you please replace the full try..else to:
```
if not isinstance(extra_opts, dict):
    extra_opts = {'paths': extra_opts}

env_var_opts.update(extra_opts)

if 'paths' not in env_var_opts:
    error_msg = f"'paths' key not set for ${env_var_name} in '{ec_param}' easyconfig parameter"
    raise EasyBuildError(error_msg)
```

Reply:
Thank you for the comment. This looks much cleaner.